modules/tracker/rbt/include/visp3/rbt/vpRBTrackingResult.h

```diff
@@ -0,0 +1,286 @@
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See https://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+/*!
+  \file vpRBTrackingResult.h
+  \brief Structure containing information about the tracking process for a given iteration
+*/
+#ifndef VP_RB_TRACKING_RESULT_H
+#define VP_RB_TRACKING_RESULT_H
+
+#include <visp3/core/vpConfig.h>
+
+
+#include <vector>
+#include <visp3/core/vpColVector.h>
+#include <visp3/core/vpHomogeneousMatrix.h>
+
+#include <visp3/rbt/vpRBTrackingTimings.h>
+#include <visp3/rbt/vpRBFeatureTracker.h>
+
+#if defined(VISP_HAVE_NLOHMANN_JSON)
+#include VISP_NLOHMANN_JSON(json.hpp)
+#endif
+
+BEGIN_VISP_NAMESPACE
+
+class VISP_EXPORT vpRBFeatureResult
+{
+
+public:
+  vpRBFeatureResult() = default;
+  void onIter(vpRBFeatureTracker &tracker)
+  {
+    m_numFeatures.push_back(tracker.getNumFeatures());
+    double w = tracker.getVVSTrackerWeight();
+    if (!vpMath::isFinite(w)) {
+      w = 0.0;
+    }
+    m_overallWeight.push_back(w);
+    m_error.push_back(tracker.getWeightedError());
+    m_JTJ.push_back(tracker.getLTL());
+    m_JTR.push_back(tracker.getLTR());
+  }
+
+  std::vector<unsigned int> getNumFeatures() const { return m_numFeatures; }
+  std::vector<double> getWeight() const { return m_overallWeight; }
+
+  std::vector<vpColVector> getErrors() const { return m_error; }
+  std::vector<vpMatrix> getJTJs() const { return m_JTJ; }
+  std::vector<vpColVector> getJTRs() const { return m_JTR; }
+
+
+#ifdef VISP_HAVE_NLOHMANN_JSON
+  inline friend void from_json(const nlohmann::json &j, vpRBFeatureResult &result);
+  inline friend void to_json(nlohmann::json &j, const vpRBFeatureResult &result);
+#endif
+
+private:
+  std::vector<unsigned int> m_numFeatures;
+  std::vector<double> m_overallWeight;
+
+  std::vector<vpColVector> m_error;
+  std::vector<vpMatrix> m_JTJ;
+  std::vector<vpColVector> m_JTR;
+};
+
+
+enum vpRBTrackingStoppingReason
+{
+  MAX_ITERS = 0, // Reached maximum number of iterations
+  CONVERGENCE_CRITERION = 1, //! Convergence criterion was met, early stopping
+  OBJECT_NOT_IN_IMAGE = 2, //! Object is not in image, thus tracking was not performed
+  NOT_ENOUGH_FEATURES = 3, //! Not enough features to correctly track the object
+  EXCEPTION = 4,
+  INVALID_REASON = 5 //! This should not happen
+};
+
+
+
+class VISP_EXPORT vpRBTrackingResult
+{
+public:
+
+  vpRBTrackingResult() : m_odometryMetric(0.0), m_odometryThreshold(0.0)
+  {
+    m_odometryMotion = vpHomogeneousMatrix();
+  }
+
+  unsigned int numIterations() const { return m_cMos.size(); }
```

Comment:
Instead of `numIterations()` I suggest to rename this method to `getNumIterations()`